instrumentation/kafka-connect-metrics-1.0.0/src/main/java/com/nr/instrumentation/kafka/connect/KafkaConnectMetricsReporter.java

```diff
@@ -0,0 +1,176 @@
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+
+public class KafkaConnectMetricsReporter implements org.apache.kafka.common.metrics.MetricsReporter {
+
+    private static final String KAFKA_CONNECT_METRICS_EVENT_NAME = "KafkaConnectMetrics";
+
+    private static final String METRIC_PREFIX = "Kafka/Connect/";
+
+    private static final boolean KAFKA_METRICS_DEBUG = NewRelic.getAgent().getConfig().getValue("kafka.metrics.debug.enabled", false);
+
+    private static final boolean METRICS_AS_EVENTS = NewRelic.getAgent().getConfig().getValue("kafka.metrics.as_events.enabled", false);
+
+    private static final long REPORTING_INTERVAL_IN_SECONDS = NewRelic.getAgent().getConfig().getValue("kafka.metrics.interval", 30);
+
+    private static final ScheduledThreadPoolExecutor EXECUTOR = new ScheduledThreadPoolExecutor(1, buildThreadFactory());
+
+    private static final AtomicBoolean INITIALIZED = new AtomicBoolean(false);
+
+    // key is the metric/event name
+    private final Map<String, KafkaMetric> metrics = new ConcurrentHashMap<>();
+
+    private ScheduledFuture<?> scheduledFuture;
+
+    // the metrics are shared between Workers. So only one reporter needs to be attached.
+    public static void initialize(ConnectMetrics.MetricGroup metricGroup) {
+        if (metricGroup != null && metricGroup.metrics() != null &&
+                !INITIALIZED.getAndSet(true)) {
+            metricGroup.metrics().addReporter(new KafkaConnectMetricsReporter());
+        }
+    }
+
+    @Override
+    public void init(final List<KafkaMetric> initMetrics) {
+        for (KafkaMetric kafkaMetric : initMetrics) {
+            String metricName = getMetricName(kafkaMetric);
+            if (KAFKA_METRICS_DEBUG) {
+                NewRelic.getAgent().getLogger().log(Level.FINEST,
+                        "init(): {0} = {1}", metricName, kafkaMetric.metricName());
+            }
+            metrics.put(metricName, kafkaMetric);
+        }
+        scheduledFuture = EXECUTOR.scheduleAtFixedRate(
+                this::harvest, 0L, REPORTING_INTERVAL_IN_SECONDS, TimeUnit.SECONDS);
+    }
+
+    // default visibility for testing
+    void harvest() {
+        try {
+            Map<String, Object> eventData = METRICS_AS_EVENTS ? new HashMap<>() : Collections.emptyMap();
+            for (final Map.Entry<String, KafkaMetric> metric : metrics.entrySet()) {
+                Object metricValue = metric.getValue().metricValue();
+                if (metricValue instanceof Double) {
```

Comment:
Why not use `metricValue instanceof Number`? There could be missing metrics that cast to an integer.

Reply:
In addition you can then get your float value like this: `final float value = ((Number) metricValue).floatValue();`

Reply:
This was copied from the original Kafka metrics instrumentation.
But that is a good change. I checked and apparently they do have integer metrics.